src/OrbitBase/include/OrbitBase/Typedef.h

```diff
@@ -135,19 +135,31 @@ class Typedef {
 template <typename Tag>
 class Typedef<Tag, void> {};
 
-// When the `Tag` inherits from the struct, the `Typedef<Tag, T>` supports `operator+`
-struct PlusTag {};
+// When the `Tag` inherits from the struct, the `Typedef<Tag, T>` supports `operator+` with other
+// `Typedef<OtherSummandTag, T>`. The result is wrapped into `Typedef<Tag, T>`.
```

Comment:
I think this should mention that if a user doesn't need cross-type summation they can set `OtherSummandTag` to `Tag`.

Nit: In "The result is wrapped into `Typedef<Tag, T>`.", the `T` is in general different to the previous `T`